libs/libarchfpga/src/switchblock_types.h

```diff
@@ -24,6 +26,20 @@ enum e_side : unsigned char {
     NUM_3D_SIDES = 6,
 };
 
+const std::unordered_map<char, e_side> CHAR_SIDE_MAP = {
+    {'T', TOP},
```

Comment:
I'd comment what these are.  E.g. "At the intersection of routing channels, left, right, top and bottom specify the x- and y-directed channels while above and under specify the switch block wires one a layer above or below the current one.  above and below are only used for multi-layer FPGAs."